Microsoft.NET.Build.Containers/Tasks/CreateNewImageToolTask.cs

```diff
@@ -13,7 +13,7 @@ namespace Microsoft.NET.Build.Containers.Tasks;
 /// <summary>
 /// This task will shell out to the net7.0-targeted application for VS scenarios.
 /// </summary>
-public partial class CreateNewImage : ToolTask
+public partial class CreateNewImage : ToolTask, ICancelableTask
```

Comment:
`ToolTask` already implements `ICancelableTask` so this shouldn't be necessary, but the default ToolTask behavior is to just rudely kill the process. Does that work ok for us?

Reply:
just discovered the same
- i kept `ICancelableTask` for readability only, so noone asks why it's not cancelable.
- i just discovered same about `ToolTask` and that's not good. This PR adds only very basic cancelation, and nothing really depends on it being triggered now. However, built-in methods might have better behavior with disposing / removing resources before cancellation happens. Ideally, ToolTask should send terminate a process (at least attempt to), before killing the process.

Reply:
i see that `ToolTask` allows to override `Cancel` and `Execute`, but it feels like too much duplicate logic just to handle termination differently. Ideally `TerminateToolProcess` is the place to customize, but it's private now.

Also, adding an option to terminate the process might be a nice improvement overall for any `ToolTask` implementation.

Reply:
> Also, adding an option to terminate the process might be a nice improvement overall for any `ToolTask` implementation.

What do you mean by "terminate"? I don't think Windows or .NET has any equivalent of `SIGTERM`.

Reply:
In windows there is a way to send Ctrl+C using https://learn.microsoft.com/en-us/windows/console/generateconsolectrlevent. 

https://stackoverflow.com/questions/813086/can-i-send-a-ctrl-c-sigint-to-an-application-on-windows has some details.

Reply:
Note: System.CommandLine also adds handler for https://learn.microsoft.com/en-us/dotnet/api/system.appdomain.processexit?view=net-7.0, but it is not called when process is killed either.

https://github.com/dotnet/command-line-api/blob/1bc03fed713b0b10455942d9f8eeab371d519cba/src/System.CommandLine/Builder/CommandLineBuilderExtensions.cs#L34-L121